test/metabase/pulse/render/body_test.clj

```diff
@@ -621,6 +622,50 @@
                             doc)]
             (is (not (render-error? pulse-body)))))))))
 
+(def ^:private funnel-rows
+  [["cart" 1500]
+   ["checkout" 450]
+   ["homepage" 10000]
+   ["product_page" 5000]
+   ["purchase" 225]])
+
+(tx/defdataset funnel-data
+  [["stages"
+    [{:field-name "stage", :base-type :type/Text}
+     {:field-name "count", :base-type :type/Quantity}]
+    funnel-rows]])
+
+(deftest render-funnel-text-row-labels-test-2
```

Comment:
Naming nit, how about calling out the semantic distinction between the test (indexes versus keys), rather than having "test" and "test-2".

Otherwise, I think that test names should always end in `-test`, so any "suffix" should come before that.